AdyenDropIn/Modules/PreselectedPaymentMethod/PreselectedPaymentMethodViewModel.swift

```diff
@@ -0,0 +1,106 @@
+//
+// Copyright (c) 2025 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+import UIKit
+@_spi(AdyenInternal) import Adyen
+
+internal protocol PreselectedPaymentMethodRouterProtocol: AnyObject {
+    func showAllPaymentMethods()
+    func proceed(with component: any PaymentComponent)
+}
+
+internal protocol PreselectedPaymentMethodViewModelProtocol {
+    var paymentMethodView: UIViewController { get }
+}
+
+internal class PreselectedPaymentMethodViewModel: PreselectedPaymentMethodViewModelProtocol, PreselectedPaymentMethodComponentDelegate {
+
+    // MARK: - Properties
+
+    private weak var router: PreselectedPaymentMethodRouterProtocol?
+    private let preselectedPaymentMethodComponent: PreselectedPaymentMethodComponent
+
+    // MARK: - Initializers
+
+    internal init(
+        router: PreselectedPaymentMethodRouterProtocol,
+        component: PaymentComponent,
+        title: String,
+        style: FormComponentStyle,
+        listItemStyle: ListItemStyle,
+        localizationParameters: LocalizationParameters?
+    ) {
+        self.router = router
+        self.preselectedPaymentMethodComponent = PreselectedPaymentMethodComponent(
+            component: component,
+            title: title,
+            style: style,
+            listItemStyle: listItemStyle
+        )
+        self.preselectedPaymentMethodComponent.localizationParameters = localizationParameters
+        self.preselectedPaymentMethodComponent.delegate = self
+    }
+
+    // MARK: - PreselectedPaymentMethodViewModelProtocol
+
+    internal var paymentMethodView: UIViewController {
```

Comment:
```suggestion
    internal var paymentMethodViewController: UIViewController {
```

Full name would be more readable.

Reply:
In next PR I am creating this is gone, so you won't see this property any more. Just something temporary